python/python/glide/async_commands/core.py

```diff
@@ -324,6 +341,61 @@ def get_cmd_args(self) -> List[str]:
         return [self.cmd_arg] if self.value is None else [self.cmd_arg, self.value]
 
 
+class ExpiryGetEx:
+    """GetEx option: Represents the expiry type and value to be executed with "GetEx" command."""
+
+    def __init__(
+        self,
+        expiry_type: ExpiryTypeGetEx,
+        value: Optional[Union[int, datetime, timedelta]],
+    ) -> None:
+        """
+        Args:
+            - expiry_type (ExpiryType): The expiry type.
+            - value (Optional[Union[int, datetime, timedelta]]): The value of the expiration type. The type of expiration
+                determines the type of expiration value:
+                - SEC: Union[int, timedelta]
+                - MILLSEC: Union[int, timedelta]
+                - UNIX_SEC: Union[int, datetime]
+                - UNIX_MILLSEC: Union[int, datetime]
+                - PERSIST: Type[None]
```

Comment:
This is kind of ick. 
This means the user needs to set the expiration type when selecting PERSIST.  
It would be much better if we could ignore this variable completely when selecting PERSIST.

Reply:
Maybe include constructors for each type, so we don't have this in the constructor (we favour compile time errors, not runtime errors): `value: Optional[Union[int, datetime, timedelta]]`